### PR TITLE
Task accordion and sub tasks

### PR DESCRIPTION
**Pull request by erikguntner**
https://github.com/hackforla/HomeUniteUs/assets/27253583/a5629fd5-fabe-41b4-bad0-7aa3b2461c39

This PR includes the following changes:

- Adds task accordion components to the My Tasks section on the dashboard homepage
- Task accordion components can be opened by clicking on them, displaying a list of what I've called subtasks.
- Task accordions display a count of how many subtasks have been completed
- Task accordions render different elements based on the status of the task
- Task accordion with the "in progress" task is expanded on page load
- Subtasks display a title, description, and icon that changes based on the current status.
- Subtasks also display either a button or message based on the status

Still to do:

- Components are using dummy data so this will need to be replaced with an API call when the endpoints are in place.

Closes [568](https://github.com/hackforla/HomeUniteUs/issues/568) Closes [569](https://github.com/hackforla/HomeUniteUs/issues/569)

Below are the Typescript types representing the general structure:

```
// The status of a task
export type TaskStatus = 'in-progress' | 'complete' | 'locked';

export interface Task {
  id: number;
  title: string;
  status: TaskStatus;
  subTasks: SubTask[]; // array of subtasks
}

export interface SubTask {
  id: number;
  title: string;
  status: TaskStatus;
  description: string;
  buttonText: string;
  url: string;
}
```
An example of a Task object:
```
  {
    id: 1,
    title: 'Application and Onboarding',
    status: 'in-progress',
    subTasks: [
      {
        id: 1,
        title: 'Application',
        status: 'complete',
        description:
          'Start your guest application to move on to the next step.',
        buttonText: 'Start application',
        url: '/guest-application',
      },
      {
        id: 2,
        title: 'Coordinator Interview',
        status: 'in-progress',
        description: 'Meet with your Coordinator to share more about yourself.',
        buttonText: 'Schedule interview',
        url: '/schedule',
      },
      {
        id: 3,
        title: 'Training Session',
        status: 'locked',
        description:
          'Complete a training session to prepare you for the host home experience.',
        buttonText: 'Schedule training',
        url: '/schedule',
      },
    ],
  },
```

My idea for this structure came partly from the design requirements but also from issue [391](https://github.com/hackforla/HomeUniteUs/issues/391) and how the state machine and events will move the user through this process. The main tasks and subtasks all have IDs that correspond to various states in the state machine and their status would then be updated as different events occur. For example, after a user submits their application and it's approved, an event would fire changing the status of the `submit_application` task to "complete" and simultaneously changing the status of ` interview_with_coordinator` to  "in-progress".

I'd love to get your feedback on the structure and my current approach, as well as what can be done to improve it. Let me know if the properties and names need to be changed, what could be added or removed, or if this is just way off.  I hope it can start a discussion about how we want to structure the data on the backend and what is needed on the frontend. Thanks!

```mermaid
erDiagram
    USER ||--|{ TASK : has
    TASK {
        int id PK
        int userId FK
        string title
        int statusId FK 
    }
    TASK ||--|{ SUB_TASK: contains
    TASK ||--|| STATUS: has
    SUB_TASK ||--|| STATUS: has
    SUB_TASK{
        int id PK
        int taskId FK
        string title
        string description
        int statusId FK
        string buttonText 
        url string
    }
    STATUS {
       int id PK
       string status "locked, inProgress, complete" 
    }
```

### CODE REVIEW

File: app/src/views/GuestApplicationTracker.tsx [hunk 1]
```diff
@@ -1,5 +1,97 @@
...
+export interface Task {
+  id: TaskIds;
+  title: string;
+  status: TaskStatus;
+  subTasks: SubTask[];
+}
+
+const tasks: Task[] = [
+  {
+    id: 'application_and_onboarding',
```
Comment (paulespinosa) [inline app/src/views/GuestApplicationTracker.tsx:36]: Will it be ok if the `id` is an integer?

File: app/src/views/GuestApplicationTracker.tsx [hunk 2]
```diff
@@ -1,5 +1,97 @@
...
+    title: 'Application and Onboarding',
+    status: 'in-progress',
+    subTasks: [
+      {
+        id: 'submit_application',
```
Comment (paulespinosa) [inline app/src/views/GuestApplicationTracker.tsx:41]: Same question about `id` being integer here and other places as well.

File: app/src/views/GuestApplicationTracker.tsx [hunk 3]
```diff
@@ -1,5 +1,97 @@
...
+  status: TaskStatus;
+  subTasks: SubTask[];
+}
+
+const tasks: Task[] = [
+  {
+    id: 'application_and_onboarding',
+    title: 'Application and Onboarding',
+    status: 'in-progress',
+    subTasks: [
```
Comment (paulespinosa) [inline app/src/views/GuestApplicationTracker.tsx:39]: Do the subtasks need a link or other data to let the FE know where to send the user to complete the subtask?

File: app/src/components/dashbaord/DashboardTask.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  console.log('do something');
+};
+
+interface TaskInfo {
+  title: string;
+  onClick?: () => void;
+}
+
+const taskInfo: {[key in SubTaskIds]: TaskInfo} = {
+  submit_application: {
```
Comment (paulespinosa) [inline app/src/components/dashbaord/DashboardTask.tsx:17]: As I understand the set of tasks is fixed for the MVP. Is that correct?

**Review by paulespinosa — COMMENTED** [review]
It's looking good! I just need to do a little more background research on the feature to know more about the number of tasks, subtasks, etc. to expect. Thank you!

File: app/src/views/GuestApplicationTracker.tsx (hunk 3 above)
Comment (erikguntner) [reply app/src/views/GuestApplicationTracker.tsx:39]: @paulespinosa Yes, if that's something we could keep in the db that would be great.

File: app/src/components/dashbaord/DashboardTask.tsx (hunk 4 above)
Comment (erikguntner) [reply app/src/components/dashbaord/DashboardTask.tsx:17]: @paulespinosa I'm not sure about the scope of the tasks for the MVP right now.

**Review by mira-kine — APPROVED** [review]
Feels clear to me in terms of data types and how it is organized. My only question is regarding rejection/withdraw events (per issue #391): would taskId just go back to host_matching again, and reset statuses? As in, do we need any way to keep track of the rejection/withdraw events?

**Review by agosmou — APPROVED** [review]
UI looks great! I'm working through the design of the endpoints, so I'll use some of this info to navigate the design.

**Pull request merged**